src/main/java/com/example/glovodb/controller/ProductController.java

```diff
@@ -0,0 +1,36 @@
+package com.example.glovodb.controller;
+
+import com.example.glovodb.model.Product;
+import com.example.glovodb.service.ProductService;
+
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+
+    private final ProductService productService;
+
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+    @GetMapping
+    public List<Product> getAllProduct(){
+        return productService.getAllProduct();
+    }
+    @GetMapping("/{id}")
+    public Optional<Product> getProductById(@PathVariable Integer id){
+        return productService.getProductById(id);
+    }
+    @PostMapping
+    public Product addProduct(@RequestBody Product product){
+        return productService.addProduct(product);
+    }
+    @PostMapping("/orders/{id}")
```

Comment:
order contains product  so this controller method need to be in order controller